src/components/orgunits/OrgUnitDialog.js

```diff
@@ -130,75 +85,78 @@ export class OrgUnitDialog extends Component {
-                                {i18n.t('No data found for this period.')}
-                            </div>
-                        )}
+            <Modal position="middle" onClose={this.onClose}>
+                <ModalTitle>{name}</ModalTitle>
+                <ModalContent>
+                    <div className={styles.orgunit}>
+                        <div className={styles.metadata}>
+                            <h3>{i18n.t('Parent unit')}</h3>
+                            {parent.name}
+                            <h3>{i18n.t('Code')}</h3>
+                            {code}
+                            <h3>{i18n.t('Groups')}</h3>
+                            {groups.map(group => (
+                                <div key={group.id}>{group.name}</div>
+                            ))}
+                        </div>
+                        <div className={styles.data}>
+                            <PeriodSelect
+                                periodType={periodType}
+                                period={period}
+                                onChange={this.onPeriodChange}
+                            />
+                            {data && data.length ? (
+                                <Table>
+                                    <TableHead>
+                                        <TableRowHead>
+                                            <TableCellHead dense>
+                                                {i18n.t('Data element')}
+                                            </TableCellHead>
+                                            <TableCellHead
+                                                dense
+                                                className={styles.right}
+                                            >
+                                                {i18n.t('Value')}
+                                            </TableCellHead>
+                                        </TableRowHead>
+                                    </TableHead>
+                                    <TableBody>
+                                        {data.map(({ id, name, value }) => (
+                                            <TableRow key={id}>
+                                                <TableCell dense>
+                                                    {name}
+                                                </TableCell>
+                                                <TableCell
+                                                    dense
+                                                    className={styles.right}
+                                                >
+                                                    {value}
+                                                </TableCell>
+                                            </TableRow>
+                                        ))}
+                                    </TableBody>
+                                </Table>
```

Comment:
Random suggestion - can we move the table out into its own component?

Reply:
Done!